tests/test_multi.py

```diff
@@ -36,7 +36,7 @@ class N:
         class A:
             a: N
 
-        assert multi.string("a { b = 1\nc = 2 }").string("a { c = 3\nd = 4 }").on(
+        assert multi.string("a { b = 1,\nc = 2 }").string("a { c = 3,\nd = 4 }").on(
```

Comment:
comma should not be required in the spec, likely the issue is with windows using `\r\n` and `\n`, can you try with that?

Reply:
Even if I try with \r\n, the result is the same:

```
FAILED tests/test_multi.py::TestMulti::test_complex - pyparsing.exceptions.ParseSyntaxException: , found '='  (at char 13), (line:2, col:3)
```

Reply:
Make sure you did not upgrade vyparsing by mistake: https://github.com/zifeo/dataconf/blob/main/poetry.lock#L449

Reply:
I have pyparsing 3.1.1.

I'm going to debug it in details hopefully tonight.

Reply:
Very strange. This fails:

`        assert multi.string("a { b = 1\nc = 2 }").string("a { c = 3\nd = 4 }").on(
            A
        ) == A(a=N(b=1, c=3, d=4))`

But this, basically the same assertion, but passes:

`        assert multi.string("a { b = 1\nc = 2 }").string("a { d = 4\nc = 3 }").on(
            A
        ) == A(a=N(b=1, c=3, d=4))`

Difference is that in the second string c and d swapped.

Reply:
there are a lot of weirdness in the latest pyhocon change (that bumped pyparsing): https://github.com/chimpler/pyhocon/issues?q=is%3Aissue+is%3Aopen+0.3.60, I would stick at pyparsing v2 and those issues should be gone

Reply:
I can confirm that my tests are passing with pyhocon==0.3.59 and pyparsing==2.4.7. Should dataconf depend on these versions instead?

Reply:
Yes please and we can merge your PR 👍

Reply:
Fine! is it enough to update version in pyproject.toml?

Reply:
Updated also poetry.lock